app/controllers/api/v2/apps_controller.rb

```diff
@@ -85,6 +85,44 @@ def increase_log_count
     }, status: :ok
   end
 
+  def update_barito_app
+    valid, error_response = validate_required_keys(
+      [:app_group_secret, :app_name])
+    render json: error_response, status: error_response[:code] and return unless valid
+
+    app_group = AppGroup.find_by(secret_key: params[:app_group_secret])
+    if app_group.blank? || !app_group.available?
+      render json: {
+        success: false,
+        errors: ['AppGroup not found or inactive'],
+        code: 404
+      }, status: :not_found and return
+    end
+    
+    app = BaritoApp.find_by(
+      app_group: app_group,
+      name: params[:app_name]
+    )
+    if app.blank? || !app.available?
+      render json: {
+        success: false,
+        errors: ['App/Release not found or inactive'],
+        code: 404
+      }, status: :not_found and return
+    else
+      app.update(
```

Comment:
For the invalid params, we already capture the validation on line  89 - 91.
For the one of the them is not changed, by default we will provide the default value on gopaysh and if there is no changes then it will update with the same value.